api-metastore/src/main/java/org/zalando/nakadi/controller/SchemaController.java

```diff
@@ -53,6 +55,32 @@ public SchemaController(final SchemaService schemaService, final EventTypeServic
         return status(HttpStatus.OK).build();
     }
 
+    @RequestMapping(value = "/event-types/{name}/schemas/{version}", method = RequestMethod.PUT)
+    public ResponseEntity<?> checkCompatibility(@PathVariable("name") final String name,
+                                                @PathVariable("version") final String version,
+                                    @Valid @RequestBody final EventTypeSchemaBase schema,
+                                    final Errors errors) {
+        if (errors.hasErrors()) {
+            throw new ValidationException(errors);
+        }
+
+        final EventType eventType = eventTypeService.get(name);
+        if(!schema.getType().equals(eventType.getSchema().getType())){
+            throw new SchemaValidationException("schema type cannot be different");
+        }
+
+        final var toCompareEventTypeSchema = version.equals("latest")?
+                eventType.getSchema(): schemaService.getSchemaVersion(name, version);
+        eventType.setSchema(toCompareEventTypeSchema);
+
+        final var newEventTypeBase = new EventTypeBase(eventType);
+        newEventTypeBase.setSchema(schema);
+
+        schemaService.getValidEvolvedEventType(eventType, newEventTypeBase);
```

Comment:
those spring runtime exception processing is total magic. what happens if they are not compatible?

Reply:
what error message and status code will be returned?

Reply:
Currently 200 is returned for successful evolution otherwise 422 
An example response is,
```
{
    "title": "Unprocessable Entity",
    "status": 422,
    "detail": "Failed to evolve avro schema due to [{ type:READER_FIELD_MISSING_DEFAULT_VALUE, location:/fields/0, message:zip_code }]"
}
```